src/Fantomas.Core/CodePrinter.fs

```diff
@@ -485,135 +460,24 @@ let genExpr (e: Expr) =
                     let size = getListOrArrayExprSize ctx ctx.Config.MaxArrayOrListWidth node.Elements
                     isSmallExpression size smallExpression multilineExpression ctx
             |> genNode node
-    | RecordOrAnonRecord node ->
-        let fieldsExpr = col sepNln node.Fields genRecordFieldName
-        let hasFields = List.isNotEmpty node.Fields
-
-        let isStruct =
-            match node.AnonRecord with
-            | Some info -> info.IsStruct
-            | None -> false
-
-        let smallRecordExpr =
-            onlyIf isStruct !- "struct "
-            +> genSingleTextNode node.OpeningBrace
-            +> addSpaceIfSpaceAroundDelimiter
-            +> match node.Extra with
-               | RecordNodeExtra.Inherit ie ->
-                   (genSingleTextNode ie.InheritKeyword +> sepSpace +> genInheritConstructor ie
-                    |> genNode (InheritConstructor.Node ie))
-                   +> onlyIf hasFields sepSemi
-               | RecordNodeExtra.With we -> genExpr we +> !- " with "
-               | RecordNodeExtra.None -> sepNone
-            +> col sepSemi node.Fields genRecordFieldName
-            +> addSpaceIfSpaceAroundDelimiter
-            +> genSingleTextNode node.ClosingBrace
+    | Expr.Record node ->
+        let smallRecordExpr = genSmallCopyableRecordNode node
+        let genCrampedFields = genMultilineStandardRecordExprCrampedFields node
```

Comment:
`genMultilineStandardRecordExprCrampedFields` is only used in one place. If that is the case, I prefer it to be inside the match clause body. Only re-usable functions should be outside of it.